### PR TITLE
mesa: LLVM 9.x migration

### LINKED ISSUE #4201: mesa: migrate to LLVM 9.x
mesa_swpipe retains llvm <9 dependency conflict

### PR DESCRIPTION
**Pull request by kenmays**
Reduce llvm <9 conflict issues.
Closes #4201

### CODE REVIEW

**Comment by diversys** [comment]
Does BilliardGL work after building mesa with LLVM9?

**Comment by kenmays** [comment]
haiku_BillardGL-1 75-4
@diversys - any issues on your end?

File: sys-libs/mesa/mesa-17.1.10.recipe [hunk 1]
```diff
@@ -55,13 +55,13 @@ PROVIDES_swpipe="
...
 	mesa$secondaryArchSuffix == $portVersion base
-	lib:libLLVM_7$secondaryArchSuffix
+	lib:libLLVM_9$secondaryArchSuffix
 	"
 
 BUILD_REQUIRES="
 	haiku${secondaryArchSuffix}_devel
 	devel:libz$secondaryArchSuffix
-	devel:libLLVM_7$secondaryArchSuffix
+	devel:libLLVM_9$secondaryArchSuffix
```
Comment (waddlesplash) [inline sys-libs/mesa/mesa-17.1.10.recipe]: There is still a llvm_config required below. Which one is that?

Comment (kenmays) [reply sys-libs/mesa/mesa-17.1.10.recipe]: cmd:llvm-config = llvm9-9.0.0-2

File: sys-libs/mesa/mesa-17.1.10.recipe [hunk 2]
```diff
@@ -27,6 +27,7 @@ PROVIDES="
 	"
 REQUIRES="
 	haiku$secondaryArchSuffix
+	lib:libLLVM_9$secondaryArchSuffix
```
Comment (korli) [inline sys-libs/mesa/mesa-17.1.10.recipe]: Shouldn't be needed.

Comment (kenmays) [reply sys-libs/mesa/mesa-17.1.10.recipe]: Used by Gallium osmesa lib.

Comment (korli) [reply sys-libs/mesa/mesa-17.1.10.recipe]: please move it to a subpackage.

Comment (kenmays) [reply sys-libs/mesa/mesa-17.1.10.recipe]: Fixed.

File: sys-libs/mesa/mesa-17.1.10.recipe [hunk 3]
```diff
@@ -104,6 +106,7 @@ BUILD()
 		--disable-gles2 \
 		--disable-shared-glapi \
 		--with-gallium-drivers=swrast \
+		--enable-texture-float \
```
Comment (korli) [inline sys-libs/mesa/mesa-17.1.10.recipe]: This is to document in the commit message.

Comment (waddlesplash) [reply sys-libs/mesa/mesa-17.1.10.recipe]: @kenmays you have still not documented why this was done.

Comment (kenmays) [reply sys-libs/mesa/mesa-17.1.10.recipe]: Allows use of floating point texture data.

**Comment by kenmays** [comment]
Don't promote mesa 17.x to LLVM 9.0.0. Need to review LLVM9 further.

**Pull request closed without merging**